plugin-server/src/worker/ingestion/action-matcher.ts

```diff
@@ -137,43 +139,125 @@ export class ActionMatcher {
         this.statsd = statsd
     }
 
-    /** Get all actions matched to the event. */
     public async match(event: PostIngestionEvent, elements?: Element[]): Promise<Action[]> {
-        const matchingStart = new Date()
+        // Matching bytecode first to assure returned action order does not change while awaiting.
+        const hogVMResponse = this.matchBytecode(event, elements)
+        const legacyResponse = await this.matchLegacy(event, elements)
+
+        if (
+            legacyResponse.length !== hogVMResponse.length ||
+            !legacyResponse.every((action, index) => action.id == hogVMResponse[index]?.id)
+        ) {
+            if (hogVMResponse.length > legacyResponse.length) {
+                this.statsd?.increment('action_matches_hogvm_legacy_mismatch_hogvm_matched_more')
+            } else {
+                this.statsd?.increment('action_matches_hogvm_legacy_mismatch')
```

Comment:
```suggestion                   
                this.statsd?.increment('action_matches_hogvm_legacy_mismatch_hogvm_matched_less')
```
Very wordy but this is in line with the other metric's naming